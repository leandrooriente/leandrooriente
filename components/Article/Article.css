.article {
  max-width: 700px;
  margin: 0 auto;
  font-size: 21px;
  line-height: 1.6;
}

.article a {
  color: var(--highlight-color);
  text-decoration: none;
}

.article img {
  max-width: 100%;
}

.article a:hover {
  border-bottom: 1px solid var(--highlight-color);
}

.title {
  margin-bottom: 10px;
}

.header {
  margin-bottom: 40px;
}
