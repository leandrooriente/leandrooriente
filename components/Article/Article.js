import React from 'react';
import PropTypes from 'prop-types';
import ArticleShape from 'propTypes/article';
import Title from 'components/Title';
import Date from 'components/Date';
import Comments from 'components/Comments';
import Back from './components/Back';
import Notification from './components/Notification';
import css from './Article.css';

const Article = ({ article }) => (
  <article className={css.article}>
    <header className={css.header}>
      <Back />
      <Title className={css.title}>
        { article.title }
      </Title>
      <Date
        lang={article.lang}
        date={article.date}
      />
    </header>

    <Notification date={article.date} />

    <div
      dangerouslySetInnerHTML={{__html: article.html}}  // eslint-disable-line
    />

    <Comments />

    <Back />
  </article>
);

Article.propTypes = {
  article: PropTypes.shape(ArticleShape).isRequired,
};

export default Article;
