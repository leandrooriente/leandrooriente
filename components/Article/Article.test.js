import React from 'react';
import { mount } from 'enzyme';
import articlesMock from '__mocks__/articles';
import Article from './Article';

describe('<Article />', () => {
  it('should render <Article /> component', () => {
    const wrapper = mount(
      <Article
        article={articlesMock[0]}
      />
    );

    expect(wrapper).toMatchSnapshot();
  });
});
