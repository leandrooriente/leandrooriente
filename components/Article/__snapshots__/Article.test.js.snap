// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Article /> should render <Article /> component 1`] = `
<Article
  article={
    Object {
      "author": "Test",
      "date": "10/10/2010",
      "html": "<p>Test</p>",
      "lang": "pt-br",
      "slug": "test-test",
      "thumbnail": "test.png",
      "title": "Test",
      "url": "/test/test",
    }
  }
>
  <article
    className="article"
  >
    <header
      className="header"
    >
      <Back>
        <Link
          href="/"
        >
          <a
            className="link"
            href="/"
            onClick={[Function]}
          >
            ↤ Articles
          </a>
        </Link>
      </Back>
      <Title
        className="title"
      >
        <h1
          className="title title"
        >
          Test
        </h1>
      </Title>
      <Date
        className={null}
        date="10/10/2010"
        lang="pt-br"
      >
        <div
          className="details"
        >
          October 2010
           
          <span
            className="separator"
          >
            -
          </span>
           
          pt-br
        </div>
      </Date>
    </header>
    <Notification
      date="10/10/2010"
    >
      <div
        className="notification"
      >
        Este artigo possui mais de 
        8
         anos de existência. O seu conteúdo pode estar desatualizado ou pode não representar mais minha opinião.
      </div>
    </Notification>
    <div
      dangerouslySetInnerHTML={
        Object {
          "__html": "<p>Test</p>",
        }
      }
    />
    <Comments>
      <div
        className="wrapper"
      >
        <Subtitle>
          <h3
            className="title"
          >
            Comments
          </h3>
        </Subtitle>
        <div
          id="disqus_thread"
        />
      </div>
    </Comments>
    <Back>
      <Link
        href="/"
      >
        <a
          className="link"
          href="/"
          onClick={[Function]}
        >
          ↤ Articles
        </a>
      </Link>
    </Back>
  </article>
</Article>
`;
